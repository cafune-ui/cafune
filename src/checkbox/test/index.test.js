import Checkbox from '../';
const { CheckboxGroup } = Checkbox;
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

const checkList = [
  {
    value: 'hey',
    text: '默认用法'
  },
  {
    value: 'hi',
    text: '自定义颜色',
    checkedColor: '#e27777'
  },
  {
    value: 'holla',
    text: '开启禁用',
    disabled: true
  },
  {
    value: 'bonjour',
    text: '自定义激活图标',
    icons: {
      actived: 'more'
    }
  },
  {
    value: 'conba',
    text: '开启禁用',
    disabled: true
  },
  {
    value: 'hello',
    text: '自定义激活及未激活图标',
    icons: {
      inactive: 'https://image.flaticon.com/icons/png/128/2026/2026450.png',
      actived: 'https://image.flaticon.com/icons/png/128/2026/2026462.png'
    }
  }
];

describe('Checkbox', () => {
  it('should render properly', () => {
    let check = 'hi';
    const changeFn = jest.fn();
    const wrapper = deep(
      <div>
        {checkList.map((item, ind) => (
          <Checkbox
            checked={check === item.value}
            {...item}
            handleChange={changeFn}
            key={`s-${ind}-${item.value}`}
          >
            {item.text}
          </Checkbox>
        ))}
      </div>
    );
    expect(wrapper).toMatchSnapshot();

    wrapper
      .find('.caf-checkbox')
      .last()
      .simulate('click');
    expect(changeFn).toHaveBeenCalledWith(checkList.slice(-1)[0].value);

    check = 'bonjour';
    wrapper.render(
      <div>
        {checkList.map((item, ind) => (
          <Checkbox
            checked={check === item.value}
            {...item}
            handleChange={changeFn}
            key={`s-${ind}-${item.value}`}
          >
            {item.text}
          </Checkbox>
        ))}
      </div>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render checkbox group', () => {
    const changeFn = jest.fn();
    const checks = ['hey', 'bonjour'];
    const wrapper = deep(
      <CheckboxGroup model={checks} handleChange={changeFn} maxSelect={3}>
        {checkList.map((item, ind) => (
          <Checkbox key={`g-${ind}-${item.value}`} {...item}>
            {item.text}
          </Checkbox>
        ))}
      </CheckboxGroup>
    );

    wrapper
      .find('.caf-checkbox')
      .last()
      .simulate('click');
    expect(changeFn).toHaveBeenCalledTimes(1);

    wrapper
      .find('.caf-checkbox')
      .first()
      .simulate('click');
    expect(changeFn).toHaveBeenCalledTimes(2);

    wrapper.render(
      <CheckboxGroup
        disabled
        model
        handleChange={changeFn}
        maxSelect={checkList.length}
      >
        {checkList.map((item, ind) => (
          <Checkbox
            key={`g-${ind}-${item.value}`}
            {...item}
            checked={item.value === 'conba'}
          >
            {item.text}
          </Checkbox>
        ))}
      </CheckboxGroup>
    );
    wrapper
      .find('.caf-checkbox')
      .first()
      .simulate('click');
    expect(changeFn).toHaveBeenCalledTimes(2);

    wrapper.render(
      <CheckboxGroup
        model={checkList.map(item => item.value)}
        maxSelect={checkList.length}
      >
        <Checkbox value="fsdfa">fadsfadf</Checkbox>
        {checkList.map((item, ind) => (
          <Checkbox key={`g-${ind}-${item.value}`} {...item}>
            {item.text}
          </Checkbox>
        ))}
      </CheckboxGroup>
    );
    wrapper
      .find('.caf-checkbox')
      .first()
      .simulate('click');
    expect(changeFn).toHaveBeenCalledTimes(2);
  });
});
