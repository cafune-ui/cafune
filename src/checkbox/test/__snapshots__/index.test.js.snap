// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "checked": false,
          "handleChange": [MockFunction],
          "key": "s-0-hey",
          "text": "默认用法",
          "value": "hey",
        },
        "children": Array [
          "默认用法",
        ],
        "key": "s-0-hey",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": true,
          "checkedColor": "#e27777",
          "handleChange": [MockFunction],
          "key": "s-1-hi",
          "text": "自定义颜色",
          "value": "hi",
        },
        "children": Array [
          "自定义颜色",
        ],
        "key": "s-1-hi",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "disabled": true,
          "handleChange": [MockFunction],
          "key": "s-2-holla",
          "text": "开启禁用",
          "value": "holla",
        },
        "children": Array [
          "开启禁用",
        ],
        "key": "s-2-holla",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "handleChange": [MockFunction],
          "icons": Object {
            "actived": "more",
          },
          "key": "s-3-bonjour",
          "text": "自定义激活图标",
          "value": "bonjour",
        },
        "children": Array [
          "自定义激活图标",
        ],
        "key": "s-3-bonjour",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "disabled": true,
          "handleChange": [MockFunction],
          "key": "s-4-conba",
          "text": "开启禁用",
          "value": "conba",
        },
        "children": Array [
          "开启禁用",
        ],
        "key": "s-4-conba",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "handleChange": [MockFunction],
          "icons": Object {
            "actived": "https://image.flaticon.com/icons/png/128/2026/2026462.png",
            "inactive": "https://image.flaticon.com/icons/png/128/2026/2026450.png",
          },
          "key": "s-5-hello",
          "text": "自定义激活及未激活图标",
          "value": "hello",
        },
        "children": Array [
          "自定义激活及未激活图标",
        ],
        "key": "s-5-hello",
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;

exports[`Checkbox should render properly 2`] = `
RenderContext {
  "0": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": Object {
          "checked": false,
          "handleChange": [MockFunction] {
            "calls": Array [
              Array [
                "hello",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "key": "s-0-hey",
          "text": "默认用法",
          "value": "hey",
        },
        "children": Array [
          "默认用法",
        ],
        "key": "s-0-hey",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "checkedColor": "#e27777",
          "handleChange": [MockFunction] {
            "calls": Array [
              Array [
                "hello",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "key": "s-1-hi",
          "text": "自定义颜色",
          "value": "hi",
        },
        "children": Array [
          "自定义颜色",
        ],
        "key": "s-1-hi",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "disabled": true,
          "handleChange": [MockFunction] {
            "calls": Array [
              Array [
                "hello",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "key": "s-2-holla",
          "text": "开启禁用",
          "value": "holla",
        },
        "children": Array [
          "开启禁用",
        ],
        "key": "s-2-holla",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": true,
          "handleChange": [MockFunction] {
            "calls": Array [
              Array [
                "hello",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "icons": Object {
            "actived": "more",
          },
          "key": "s-3-bonjour",
          "text": "自定义激活图标",
          "value": "bonjour",
        },
        "children": Array [
          "自定义激活图标",
        ],
        "key": "s-3-bonjour",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "disabled": true,
          "handleChange": [MockFunction] {
            "calls": Array [
              Array [
                "hello",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "key": "s-4-conba",
          "text": "开启禁用",
          "value": "conba",
        },
        "children": Array [
          "开启禁用",
        ],
        "key": "s-4-conba",
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "checked": false,
          "handleChange": [MockFunction] {
            "calls": Array [
              Array [
                "hello",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "icons": Object {
            "actived": "https://image.flaticon.com/icons/png/128/2026/2026462.png",
            "inactive": "https://image.flaticon.com/icons/png/128/2026/2026450.png",
          },
          "key": "s-5-hello",
          "text": "自定义激活及未激活图标",
          "value": "hello",
        },
        "children": Array [
          "自定义激活及未激活图标",
        ],
        "key": "s-5-hello",
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": "div",
  },
  "length": 1,
}
`;
